gusto/timeloop.py

```diff
@@ -152,6 +161,14 @@ def run(self, t, tmax, pickup=False):
 
 class AdvectionTimestepper(BaseTimestepper):
 
+    def __init__(self, state, advection_dict, diffusion_dict, physics_list=None):
+
+        super(AdvectionTimestepper, self).__init__(state, advection_dict, diffusion_dict)
+        if physics_list is not None:
+            self.physics_list = physics_list
+        else:
+            self.physics_list = []
```

Comment:
This code is replicated in both the `TimeStepper` and `AdvectionTimeStepper`. If all time steppers need it, it should be initialised in `BaseTimeStepper`, like the diffusion_dict and advection_dict arguments.